topics/request.md

```diff
@@ -1,316 +1,116 @@
+* Set the body of a request, for example, a plain text, a data object, form parameters, and so on.
 
-## Making request
-
-After client configuration we're ready to perform our first request.
-Most of the simple requests are made with pattern
-
-```kotlin
-val response = client.'http-method'<'ResponseType'>("url-string")
-```
-
-or even simpler form(due to kotlin generic type inference):
-
-```kotlin
-val response: ResponseType = client.'http-method'("url-string")
-```
-
-For example to perform a `GET` request fully reading a `String`:
-
-```kotlin
-val htmlContent = client.get<String>("https://en.wikipedia.org/wiki/Main_Page")
-// same as
-val content: String = client.get("https://en.wikipedia.org/wiki/Main_Page")
-```
-
-And in the case you are interested in the raw bits, you can read a `ByteArray`:
-
-```kotlin
-val channel: ByteArray = client.get("https://en.wikipedia.org/wiki/Main_Page")
-```
-
-Or get full [HttpResponse](https://api.ktor.io/%ktor_version%/io.ktor.client.statement/-http-response/index.html):
+The main way for making HTTP requests is the [request](https://api.ktor.io/%ktor_version%/io.ktor.client.request/request.html) function that takes a URL as a parameter. Inside this function, you can configure various request parameters: specify an HTTP method, add headers, specify the request body and content type, and so on. These parameters are exposed by the [HttpRequestBuilder](https://api.ktor.io/%ktor_version%/io.ktor.client.request/-http-request-builder/index.html) class.
 
 ```kotlin
-val response: HttpResponse = client.get("https://en.wikipedia.org/wiki/Main_Page")
 ```
+{src="snippets/_misc_client/RequestMethodWithoutParams.kt" interpolate-variables="true" disable-links="false"}
 
-The [HttpResponse](https://api.ktor.io/%ktor_version%/io.ktor.client.statement/-http-response/index.html) is downloaded in memory by default. To learn how to download response partially or work with a stream data consult with the [Streaming](streaming.md) section.
+Note that this function allows you to [receive a response](response.md) in various ways. In this example, we receive a response as an `HttpResponse` object.
 
-And even your data class using [Json](json-feature.md) feature:
+> `request` is a suspending function, so requests should be executed only from a coroutine or another suspend function. You can learn more about calling suspending functions from [Coroutines basics](https://kotlinlang.org/docs/coroutines-basics.html).
 
-```kotlin
-@Serializable
-data class User(val id: Int)
 
-val response: User = client.get("https://myapi.com/user?id=1")
-```
-
-Please note that some of response types are `Closeable` and can hold resources.
+## Set request parameters {id="parameters"}
+In this section, we'll see on how to specify various request parameters, including an HTTP method, headers, and cookies. If you need to configure some default parameters for all requests of a specific client, use the [](default-request.md) feature.
```

Comment:
If you need to configure some default params should probably be it's own heading.